FreeRTOS_DHCP.c

```diff
@@ -1007,84 +1081,84 @@
             uxRequiredBufferSize += ( 2U + uxNameLength );
         #endif /* if ( ipconfigDHCP_REGISTER_HOSTNAME == 1 ) */
 
-        /* Get a buffer.  This uses a maximum delay, but the delay will be capped
-         * to ipconfigUDP_MAX_SEND_BLOCK_TIME_TICKS so the return value still needs to
-         * be test. */
-        do
-        {
-            /* Obtain a network buffer with the required amount of storage. */
-            pxNetworkBuffer = pxGetNetworkBufferWithDescriptor( sizeof( UDPPacket_t ) + uxRequiredBufferSize, portMAX_DELAY );
-        } while( pxNetworkBuffer == NULL );
-
-        /* Leave space for the UDP header. */
-        pucUDPPayloadBuffer = &( pxNetworkBuffer->pucEthernetBuffer[ ipUDP_PAYLOAD_OFFSET_IPv4 ] );
-        pxDHCPMessage = ipCAST_PTR_TO_TYPE_PTR( DHCPMessage_IPv4_t, pucUDPPayloadBuffer );
-
-        /* Most fields need to be zero. */
-        ( void ) memset( pxDHCPMessage, 0x00, sizeof( DHCPMessage_IPv4_t ) );
-
-        /* Create the message. */
-        pxDHCPMessage->ucOpcode = ( uint8_t ) xOpcode;
-        pxDHCPMessage->ucAddressType = ( uint8_t ) dhcpADDRESS_TYPE_ETHERNET;
-        pxDHCPMessage->ucAddressLength = ( uint8_t ) dhcpETHERNET_ADDRESS_LENGTH;
-        pxDHCPMessage->ulTransactionID = FreeRTOS_htonl( EP_DHCPData.ulTransactionId );
-        pxDHCPMessage->ulDHCPCookie = ( uint32_t ) dhcpCOOKIE;
+        /* Obtain a network buffer with the required amount of storage.  It doesn't make much sense
+         * to use a time-out here, because that would cause the IP-task waiting for itself. */
```

Comment:
Grammar: _that would cause the IP-task to wait for itself_

Reply:
Changed, thanks.

Reply:
Thank you.